lua/lint/linters/glslc.lua

```diff
@@ -0,0 +1,30 @@
+-- path/to/file:line: severity: 'offending part of source code': message
+local pattern = "%s*([^:]+):(%d+): (.+): (.+: .+)"
+local groups = { "file", "lnum", "severity", "message" }
+
+local tmp_file
+if vim.fn.fn.has "win32" == 1 then
+  tmp_file = "/dev/null"
+else
+  local cache_dir = vim.fn.stdpath("cache")
+  if vim.loop.fs_stat(cache_dir) then
+    tmp_file = cache_dir .. "\\nvim-lint_glslc.spv"
```

Comment:
Does this file grow or is it overwritten with each run?

Otherwise it may be necessary to generate it on each use of the linter and clean up after.

Reply:
it is overwritten each run. SPRIV files are typically a few kB big. We could also just not enable this linter on Windows while we don't have a better solution.

`glslc` is actually a compiler (by Google distributed with the Vulkan SDK). There are also validators for `glsl` (e.g. glslValidator) but most not as convenient as this one. It might be possible to achieve that it does not require an output, but that use case is kind of strange for a compiler and the change would only be possible for future versions of the tool. I'd imagine that the requirement for an output file would exist for more tools that aren't linters in the traditional sense but rather compilers that would produce output normally

Reply:
It works at the moment, but would love to test it a few days.